src/utils/trace.ts

```diff
@@ -10,13 +11,47 @@ import { ExtensionContext, extensions } from 'vscode'
 import { TraceAttributes } from './types'
 import { Store } from './store'
 import { AppConfig } from './appConfig'
+import { log } from './logger'
 
 const extensionInfo = extensions.getExtension('Pullflow.pullflow')?.packageJSON
 
-export class Trace {
+type FakeTracer = {}
+type FakeBasicTracerProvider = {}
+type FakeAttribute = {}
+
+// Null Object Design pattern
+export function instantiatePullflowTracer(context: ExtensionContext) {
+  const { isTelemetryEnabled } = Store.get(context)
+  if (isTelemetryEnabled) {
+    return new Trace(context)
+  } else {
+    return new FakeTrace(context)
+  }
+}
+class FakeTrace {
+  tracer: FakeTracer
+  provider: FakeBasicTracerProvider
+  defaultAttributes: FakeAttribute
+
+  constructor(_context: ExtensionContext) {
+    this.provider = {}
+    this.tracer = {}
+    this.defaultAttributes = {}
+  }
+
+  dispose(): void {}
+  start({ name, attributes }: { name: string; attributes?: TraceAttributes }) {
+    console.log({ name, attributes })
```

Comment:
why are we doing console log here?

Reply:
because i need to use the attributes and name somehow otherwise it gives unused values errors.

Reply:
we can do it like one of these ways:
`start({ _name, _attributes }: { name: string; attributes?: TraceAttributes }) { ...
}`

`start({ }: { name: string; attributes?: TraceAttributes }) { ...
}`

what do you think @srzainab?